integration_tests/src/main/python/cache_test.py

```diff
@@ -51,25 +55,24 @@ def test_passing_gpuExpr_as_Expr(enableVectorizedConf):
 all_gen = [StringGen(), ByteGen(), ShortGen(), IntegerGen(), LongGen(),
            pytest.param(FloatGen(special_cases=[FLOAT_MIN, FLOAT_MAX, 0.0, 1.0, -1.0]), marks=[incompat]),
            pytest.param(DoubleGen(special_cases=double_special_cases), marks=[incompat]),
-           BooleanGen(), DateGen(), TimestampGen(), decimal_gen_default, decimal_gen_scale_precision,
-           decimal_gen_same_scale_precision, decimal_gen_64bit]
+           BooleanGen(), DateGen(), TimestampGen()] + decimal_gens
 
 @pytest.mark.parametrize('data_gen', all_gen, ids=idfn)
 @pytest.mark.parametrize('join_type', ['Left', 'Right', 'Inner', 'LeftSemi', 'LeftAnti'], ids=idfn)
-@pytest.mark.parametrize('enableVectorizedConf', enableVectorizedConf, ids=idfn)
+@pytest.mark.parametrize('enableVectorizedConf', enableVectorizedConfs, ids=idfn)
 @ignore_order
 def test_cache_join(data_gen, join_type, enableVectorizedConf):
     def do_join(spark):
         left, right = create_df(spark, data_gen, 500, 500)
         cached = left.join(right, left.a == right.r_a, join_type).cache()
         cached.count() # populates cache
         return cached
-
+    enableVectorizedConf.update(allowNegativeScaleDec)
```

Comment:
This will modify the values in the global `enableVectorizedConfs`.

```
conf = enable_vectorized_conf.copy()
conf.update(allow_negative_scale_of_decimal_conf)
assert_gpu_and_cpu_are_equal_collect(do_join, conf = conf)
```

And this goes for all of the places in this file that are using update.  For me personally I think it is a lot cleaner